pkg/speaker/subnet.go

```diff
@@ -348,11 +348,14 @@ func getNextHopFromPathAttributes(attrs []bgp.PathAttributeInterface) net.IP {
 	return nil
 }
 
-func getNextHopAttribute(neighborAddress, routeID string) string {
-	nextHop := routeID
+func (c *Controller) getNextHopAttribute(neighborAddress string) string {
+	nextHop := c.config.RouterID
 	routes, err := netlink.RouteGet(net.ParseIP(neighborAddress))
 	if err == nil && len(routes) == 1 && routes[0].Src != nil {
 		nextHop = routes[0].Src.String()
 	}
+	if c.config.NodeIP != "" && nextHop == c.config.PodIP {
+		nextHop = c.config.NodeIP
+	}
```

Comment:
We should consider deployment in duak-stack clusters.

Reply:
done